### PR TITLE
fix(modifier): ensure compatibility with ember-modifier v4

### PR DESCRIPTION
**Pull request by anehx**
This PR converts the `pikaday` modifier to use the new `ember-modifier` API introduced in [`v3.2.0`](https://github.com/ember-modifier/ember-modifier/releases/tag/v3.2.0). I migrated the modifier using [this guide here](https://github.com/ember-modifier/ember-modifier/blob/master/MIGRATIONS.md#40).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   },
   "dependencies": {
     "@embroider/addon-shim": "^0.50.2",
-    "ember-modifier": "^3.0.0"
+    "ember-modifier": "^3.2.7"
```
Comment (SergeAstapov) [inline package.json:47]: as ember-modifier@4.0.0 got published, let's include it in the range

```suggestion
    "ember-modifier": "^3.2.7 || ^4.0.0"
```

**Comment by patrickberkeley** [comment]
@anehx we're running into the compatibility issue you've fixed here. Any chance you can move this forward?

File: package.json (hunk 1 above)
Comment (anehx) [reply package.json]: Done

**Comment by tcjr** [comment]
@anehx  Is there anything blocking this from being merged? I would be happy to help in any way I can.

**Comment by anehx** [comment]
@tcjr @patrickberkeley I don't have any permissions on this repo so I can't really push this forward. Since I waited too long for a response we dropped `ember-pikaday` in favor of `ember-flatpickr`..

**Comment by renaudleo** [comment]
@SergeAstapov Anything we can do to help merging this? 🙂

**Pull request merged**